### PR TITLE
Modified the change detection validation such that segmentation metri…

### PR DESCRIPTION
**Pull request by diego-fustes**
…cs are computed instead of diarization metrics. Diarization metrics can

still be computed by means of the flag metric_type

### CODE REVIEW

File: pyannote/audio/applications/change_detection.py [hunk 1]
```diff
@@ -318,11 +335,19 @@ def main():
...
+            metric_type = SEGMENTATION_METRIC_TYPE
+
         application = SpeakerChangeDetection.from_train_dir(
             train_dir, db_yml=db_yml, training=False)
+
         application.device = device
         application.batch_size = batch_size
         application.purity = purity
+        application.metric_type = metric_type
+        application.pool_ = mp.Pool(mp.cpu_count())
```
Comment (hbredin) [inline pyannote/audio/applications/change_detection.py]: Why is this needed?  `pool_` is already instantiated in (inherited) `SpeechActivityDetection.validate_init`, isn't it?

File: pyannote/audio/applications/change_detection.py [hunk 2]
```diff
@@ -318,11 +335,19 @@ def main():
 
         purity = float(arguments['--purity'])
 
+        metric_type = arguments['--metric_type']
+        if not metric_type:
+            metric_type = SEGMENTATION_METRIC_TYPE
```
Comment (hbredin) [inline pyannote/audio/applications/change_detection.py]: I believe it should be `DIARIZATION_METRIC_TYPE` in order not to change the default behavior.

File: pyannote/audio/applications/change_detection.py [hunk 3]
```diff
@@ -65,6 +65,8 @@
   <train_dir>                Path to the directory containing pre-trained
                              models (i.e. the output of "train" mode).
   --purity=<purity>          Target segment purity [default: 0.9].
+  --metric_type=<metric_type>          Set metric_type (segmentation|diarization).
```
Comment (hbredin) [inline pyannote/audio/applications/change_detection.py]: I'd rather use `--metric` instead.

Comment (hbredin) [reply pyannote/audio/applications/change_detection.py]: That would be something like:

```
--metric=<metric>    Use "segmentation"  for segmentation and "diarization" 
                     for diarization metrics [default: diarization]
```

File: pyannote/audio/applications/change_detection.py [hunk 4]
```diff
@@ -166,19 +168,25 @@
     >>> homogeneous_segments = peak_detection.apply(raw_scores, dimension=1)
 """
 
-import torch
-import numpy as np
+import multiprocessing as mp
```
Comment (hbredin) [inline pyannote/audio/applications/change_detection.py]: See my remark below. I don't think this is needed.

File: pyannote/audio/applications/change_detection.py (hunk 2 above)
Comment (hbredin) [reply pyannote/audio/applications/change_detection.py]: But this won't be needed if you change the application docstring as suggested above.
`docopt` will automagically set it to `"diarization"` thanks to `[default: diarization]`.

File: tutorials/change-detection/README.md [hunk 5]
```diff
@@ -143,6 +143,9 @@ $ export TRAIN_DIR=${EXPERIMENT_DIR}/train/AMI.SpeakerDiarization.MixHeadset.tra
 $ pyannote-change-detection validate --purity=0.8 ${TRAIN_DIR} AMI.SpeakerDiarization.MixHeadset
 ```
 
+By default, this validation computes the segmentation metrics. You may prefer to set the flag `--metric_type=diarization`
```
Comment (hbredin) [inline tutorials/change-detection/README.md]: Let's keep "diarization" as the default behavior.

File: pyannote/audio/applications/change_detection.py (hunk 1 above)
Comment (diego-fustes) [reply pyannote/audio/applications/change_detection.py]: Actually, this is a bug. Checking `SpeechActivityDetection.validate_init`, it is returning before setting `pool_` ... Should I move that line up? Why is it returning when `self.feature_extraction_` is a `Precomputed` instance?

File: pyannote/audio/applications/change_detection.py (hunk 2 above)
Comment (diego-fustes) [reply pyannote/audio/applications/change_detection.py]: I would argue that it's better to have the "segmentation" metric as the default one, what do you think?

File: pyannote/audio/applications/change_detection.py (hunk 3 above)
Comment (diego-fustes) [reply pyannote/audio/applications/change_detection.py]: ok

File: pyannote/audio/applications/change_detection.py (hunk 1 above)
Comment (hbredin) [reply pyannote/audio/applications/change_detection.py]: Oh. Right. This is indeed a bug. I just pushed a fix.

File: pyannote/audio/applications/change_detection.py (hunk 2 above)
Comment (hbredin) [reply pyannote/audio/applications/change_detection.py]: Thinking again about this, I now agree with you. "segmentation" it is...
Still, please use the docopt way of setting the default value for `--metric`.

**Comment by diego-fustes** [comment]
I've adapted the changes. Please check it out

**Comment by hbredin** [comment]
Thanks. In the end, I chose to go with a --diarization option rather than --metric=diarization.
I thought it was clearer as, anyway, we probably won't use other kind of metrics.
